internal/app/collector/monitors/transactions_monitor.go

```diff
@@ -0,0 +1,118 @@
+package monitors
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/lidofinance/terra-fcd-rest-client/columbus-5/client"
+	"github.com/lidofinance/terra-fcd-rest-client/columbus-5/client/transactions"
+	"github.com/lidofinance/terra-fcd-rest-client/columbus-5/models"
+	"github.com/lidofinance/terra-monitors/internal/app/config"
+	"github.com/lidofinance/terra-monitors/internal/pkg/utils"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	MonitoredTransactionId MetricName = "monitored_transaction_id"
+)
+
+type TransactionsMonitor struct {
+	apiClient     *client.TerraRESTApis
+	logger        *logrus.Logger
+	addresses     []string
+	metricVectors map[MetricName]*MetricVector
+	lock          sync.RWMutex
+}
+
+func NewTransactionsMonitor(cfg config.CollectorConfig, logger *logrus.Logger) *TransactionsMonitor {
+	m := TransactionsMonitor{
+		apiClient:     utils.BuildClient(utils.SourceToEndpoints(cfg.Source), logger),
+		logger:        logger,
+		addresses:     cfg.Addresses.MonitoredAccounts,
+		metricVectors: make(map[MetricName]*MetricVector),
+		lock:          sync.RWMutex{},
+	}
+	m.logger.Infof("Initialized transactions monitor for addresses: %v", m.addresses)
+	m.InitMetrics()
+
+	return &m
+}
+
+func (m *TransactionsMonitor) Name() string {
+	return "TransactionsMonitor"
+}
+
+func (m *TransactionsMonitor) GetMetrics() map[MetricName]MetricValue {
+	return nil
+}
+
+func (m *TransactionsMonitor) GetMetricVectors() map[MetricName]*MetricVector {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+	return m.metricVectors
+}
+
+func (m *TransactionsMonitor) Handler(ctx context.Context) error {
+	tmpMetricVectors := make(map[MetricName]*MetricVector)
+	initMetrics(nil, m.providedMetricVectors(), nil, tmpMetricVectors)
+
+	for _, address := range m.addresses {
+		txs, err := m.queryTxs(ctx, address)
+		if err != nil {
+			m.logger.Errorf("Could not query transactions for %s. Error: %v", address, err)
+			return err
+		}
+
+		txID, err := fetchIDFromLastTx(txs)
+		if err != nil {
+			m.logger.Errorf("Could not fetch ID from transactions for %v. Error: %v", address, err)
+			return err
+		}
+
+		tmpMetricVectors[MonitoredTransactionId].Set(address, *txID)
+
+		m.logger.Infof("Successfully retrieved last transaction txID for %v", address)
+	}
+
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	copyVectors(tmpMetricVectors, m.metricVectors)
+
+	return nil
+}
+
+func (m *TransactionsMonitor) InitMetrics() {
+	initMetrics(nil, m.providedMetricVectors(), nil, m.metricVectors)
+}
+
+func (m *TransactionsMonitor) providedMetricVectors() []MetricName {
+	return []MetricName{MonitoredTransactionId}
+}
+
+func (m *TransactionsMonitor) queryTxs(ctx context.Context, address string) (*models.GetTxListResult, error) {
+	var limit float64 = 10
```

Comment:
hardcoded limit is not good. e.g. if we expand the `UPDATE_DATA_INTERVAL` one day to a minute or more, we could miss some blocks. I suggest moving the limit to a cfg variable or establishing a relation between `UPDATE_DATA_INTERVAL` and the monitor limit value (we know the block time and the interval). what you guys think?

Reply:
I think update data interval doesn't matter, because we only need confirmation that last transaction id changed?

Reply:
Also i picked limit 10 only because it's a minimum limit for this endpoint

Reply:
ah, sorry, I misunderstood the way it works. oll korrect